File: result/metagpt/claude/humaneval/128.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int prod_signs(int* arr, int size) {
    if (size == 0) return -32768;
    int i, sum = 0, prods = 1;
    for (i = 0; i < size; i++) {
        sum += abs(arr[i]);
        if (arr[i] == 0) prods = 0;
        if (arr[i] < 0) prods = -prods;
    }
    return sum * prods;
}

File: result/metagpt/gpt4o/humaneval/128.rs
fn prod_signs(arr: &[i32]) -> i32 {
    if arr.is_empty() {
        return -32768;
    }
    let mut sum = 0;
    let mut prods = 1;
    for &value in arr.iter() {
        sum += value.abs();
        if value == 0 {
            prods = 0;
        }
        if value < 0 {
            prods = -prods;
        }
    }
    sum * prods
}

fn main() {
    let arr = vec![1, -2, 3, 0, -4];
    let result = prod_signs(&arr);
    println!("Product of signs: {}", result);
}